src/components/AddDoubleFields/AddDoubleFields.js

```diff
@@ -27,84 +27,96 @@ const defaultProps = {
-                  {
-                    !this.props.disabled &&
-                    <div className={row('control')()}>
-                      <Control
-                        onClick={() => fields.remove(index)}
-                        icon='remove'
-                      />
-                    </div>
-                  }
-                </div>
-              </Row>
-            ))
-          }
-        </div>
-      )
-    };
-
-    render () {
-      const { config, isValidate, name, title, warningMessage } = this.props
-
-      return (
-        <div className='j-col__item'>
-          <FieldArray
-            name={`${name}`}
-            component={this.renderMembers}
-            title={title}
-            config={config}
-            isValidate={isValidate}
-            warningMessage={warningMessage}
+const passParse = config => config.type === 'number' ? parse : undefined
+
+const renderMembers = ({ fields, isValidate, config, title, warningMessage, disabled }) => {
+  return (
+    <div>
+      <div className={row()}>
+        <Label>{ title }</Label>
+        {
+          !disabled &&
+          <Control
+            onClick={() => fields.push({})}
+            icon='add'
           />
-        </div>
-      )
-    }
+        }
+      </div>
+      {
+        fields.map((member, index) => (
+          <Row className='double-fields' key={index} col>
+            <div className={row()}>
+              <div className={row('item', {pair: true})()}>
+                <Field
+                  name={`${member}.${config[0].sufix}`}
+                  type={config[0].type}
+                  component={Input}
+                  placeholder={config[0].placeholder}
+                  parse={passParse(config[0])}
+                  validate={isValidate && checkOnPattern(isValidate)}
+                  disabled={disabled}
+                />
+                {
+                  warningMessage &&
+                  <span className='j-input__warning'>{warningMessage}</span>
+                }
+              </div>
+              <div className={row('item')()}>
+                <Field
+                  name={`${member}.${config[1].sufix}`}
+                  type={config[1].type}
+                  component={Input}
+                  placeholder={config[1].placeholder}
+                  parse={passParse(config[1])}
+                  disabled={disabled}
+                />
+              </div>
+              {
+                !disabled &&
+                <div className={row('control')()}>
+                  <Control
+                    onClick={() => fields.remove(index)}
+                    icon='remove'
+                  />
+                </div>
+              }
+            </div>
+          </Row>
+        ))
+      }
+    </div>
+  )
+}
+
+class AddDoubleFields extends PureComponent {
+  constructor (props) {
+    super(props)
+
+    this.state = {}
+  }
+
+  componentDidMount () {
+    this.setState({ renderMembers })
```

Comment:
Why do we have to set this function in the state? It is not used anywhere in the component except being passed to the FieldArray as prop?

Reply:
Ha! This is the necessary evil.
In another case (without this strange hack), when component remounted, for some reason redux-form doesn't apply validation function(basically it removes `validate` property from `Field` component).

I guess it because component mounts faster then field get access to the data from the form (_read_ doesn't get members which should be rendered), and that's why the validation function doesn't fire and once info from the form(_read_ members) get inside all `Field` components renders but without `validate` property.

That's why only first time validation works, and after changing `load balancing alg` back and forward it stops working.

Because of all what is written before, we have to store `renderMembers` function into the state of the component. In this way, we will be sure that validation function will be assigned to `Field` component every time when `MultiRowFIeld` and `AddDoubleFields` component will be mounted .